src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs

```diff
@@ -235,23 +239,203 @@ public async Task BuildCompleteBuildHasNoMatchingTags()
-            var names = new List<string>();
+            var buildEvent = new AzurePipelinesController.AzureDevOpsEvent<AzurePipelinesController.AzureDevOpsMinimalBuildResource>
+            {
+                Resource = new AzurePipelinesController.AzureDevOpsMinimalBuildResource
+                {
+                    Id = 123456,
+                    Url = "test-build-url"
+                },
+                ResourceContainers = new AzurePipelinesController.AzureDevOpsResourceContainers
+                {
+                    Collection = new AzurePipelinesController.HasId
+                    {
+                        Id = "test-collection-id"
+                    },
+                    Account = new AzurePipelinesController.HasId
+                    {
+                        Id = "test-account-id"
+                    },
+                    Project = new AzurePipelinesController.HasId
+                    {
+                        Id = "test-project-id"
+                    }
+                }
+            };
+
+            var build = new JObject
+            {
+                ["_links"] = new JObject
+                {
+                    ["web"] = new JObject
+                    {
+                        ["href"] = "href"
+                    }
+                },
+                ["buildNumber"] = "123456",
+                ["definition"] = new JObject
+                {
+                    ["name"] = "path3",
+                    ["path"] = "\\test\\definition"
+                },
+                ["finishTime"] = "05/01/2008 6:00:00",
+                ["id"] = "123",
+                ["project"] = new JObject
+                {
+                    ["name"] = "test-project-name"
+                },
+                ["reason"] = "batchedCI",
+                ["requestedFor"] = new JObject
+                {
+                    ["displayName"] = "requested-for"
+                },
+                ["result"] = "failed",
+                ["sourceBranch"] = "refs/heads/sourceBranch",
+                ["startTime"] = "05/01/2008 5:00:00",
+            };
+
+            var expectedIssueOwners = new List<string>();
+            var expectedIssueNames = new List<string>();
+
+            var expectedCommentOwners = new List<string>
+            {
+                "dotnet"
+            };
+            var expectedCommentNames = new List<string>
+            {
+                "repo"
+            };
+
+            using TestData testData = SetupTestData(build, true);
+            var response = await testData.Controller.BuildComplete(buildEvent);
+            testData.VerifyAll(expectedIssueOwners, expectedIssueNames, expectedCommentOwners, expectedCommentNames);
+        }
+
+        [Test]
+        public async Task BuildCompleteUpdateExistingIssueDoesNotExist()
+        {
+            var buildEvent = new AzurePipelinesController.AzureDevOpsEvent<AzurePipelinesController.AzureDevOpsMinimalBuildResource>
```

Comment:
I'd probably "using static AzurePipelinesController".  I know it's weird, and generally evil... but I'm sort of OK with it for tests to "using static" the thing they are testing (this function is hard to read without it)